src/main/java/edu/ksu/canvas/attendance/form/CourseConfigurationValidator.java

```diff
@@ -19,6 +18,27 @@ public void validate(Object target, Errors errors) {
         if (courseConfigurationForm.getTotalClassMinutes() < courseConfigurationForm.getDefaultMinutesPerSession() && !errors.hasFieldErrors("totalClassMinutes")) {
             errors.rejectValue("defaultMinutesPerSession", "ExceedTotal.courseConfigurationForm.defaultMinutesPerSession");
         }
+
+        if(courseConfigurationForm.getGradingOn()) {
+            if (courseConfigurationForm.getAssignmentName() == null || courseConfigurationForm.getAssignmentName().length() <= 1 || courseConfigurationForm.getAssignmentName().trim().isEmpty()) {
+                errors.rejectValue("assignmentName", "Assignment Name is required.");
+            }
+
+            if (courseConfigurationForm.getAssignmentPoints() == null || courseConfigurationForm.getAssignmentPoints() <= 0) {
+                errors.rejectValue("assignmentPoints", "Total Points is a required field and must be greater than 0.");
+            }
+
+            if (courseConfigurationForm.getPresentPoints() == null || courseConfigurationForm.getTardyPoints() == null || courseConfigurationForm.getExcusedPoints() == null
+                    || courseConfigurationForm.getAbsentPoints() == null) {
+                errors.rejectValue("presentPoints", "All status point fields are required.");
+                return;
+            }
+
+            if ((courseConfigurationForm.getPresentPoints() > 100 || courseConfigurationForm.getPresentPoints() < 0) || (courseConfigurationForm.getTardyPoints() > 100 || courseConfigurationForm.getTardyPoints() < 0)
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce the number of conditional operators (7) used in the expression (maximum allowed 3). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.ome.ksu.edu/coding_rules#rule_key=squid%3AS1067)